cognite/neat/workflows/steps_registry.py

```diff
@@ -0,0 +1,136 @@
+
+
+class StepsRegistry:
+    def __init__(self, data_store_path: str | None = None):
+        self._step_classes: list[Step] = []
+        self.user_steps_path = Path(data_store_path) / "steps"
+        self.data_store_path = data_store_path
+
+    def load_step_classes(self):
+        if self._step_classes:
+            # classes already loaded - no need to reload
+            return
+        for name, step_cls in inspect.getmembers(cognite.neat.workflows.steps.lib):
+            if inspect.isclass(step_cls):
+                self._step_classes.append([name, step_cls])
+        try:
+            if self.user_steps_path:
+                self.load_custom_step_classes(self.user_steps_path, scope="global")
+        except Exception as e:
+            logging.warn(f"No user defined modules provided in {self.user_steps_path} : {e}")
+
+    def load_custom_step_classes(self, custom_steps_path: str, scope: str = "global"):
+        sys.path.append(str(custom_steps_path))
+        for step_module_name in os.listdir(custom_steps_path):
+            step_module_path = custom_steps_path / Path(step_module_name)
+            if step_module_name.startswith("__") or (
+                step_module_path.is_file() and not step_module_name.endswith(".py")
+            ):
+                continue
+
+            steps_module = importlib.import_module(step_module_name.replace(".py", ""))
+            logging.info(f"Loading user defined module from {steps_module}")
+            for name, step_cls in inspect.getmembers(steps_module):
+                base_class = getattr(step_cls, "__bases__", None)
+                if (
+                    name.startswith("__")
+                    or base_class is None
+                    or len(base_class) == 0
+                    or base_class[0].__name__ != "Step"
+                ):
+                    continue
+                logging.info(f"Loading user defined step {name} from {steps_module}")
+                if inspect.isclass(step_cls):
+                    logging.info(f"Loading user defined step {name} class {step_cls.__name__}")
+                    step_cls.scope = scope
+                    self._step_classes.append([name, step_cls])
+
+    def run_step(
+        self,
+        step_name: str,
+        flow_context: dict[str, Type[DataContract]],
+        metrics=None,
+    ) -> T_Output | None:
+        for name, step_cls in self._step_classes:
+            if name == step_name:
+                step_obj = step_cls(metrics, self.data_store_path)
+                step_obj.set_flow_context(flow_context)
+                signature = inspect.signature(step_obj.run)
+                parameters = signature.parameters
+                is_valid = True
+                input_data = []
+                missing_data = []
+                for parameter in parameters.values():
+                    try:
+                        if parameter.annotation.__name__ == "FlowMessage":
```

Comment:
This should be written more robustly as something like 
```suggestion
                        if parameter.annotation is FlowMessage:
```

I can probably do that refactor later, as I have some experience with writing this type of code.